pkg/cmd/roachtest/tests/tpcc.go

```diff
@@ -1716,9 +1720,36 @@ func runTPCCBench(ctx context.Context, t test.Test, c cluster.Cluster, b tpccBen
 					t.Fatal(err)
 				}
 				result := tpcc.NewResultWithSnapshots(warehouses, 0, snapshots)
+
+				// This roachtest uses the stats.json emitted from hdr histogram to compute Tpmc and show it in the run log
+				// Since directly emitting openmetrics and computing Tpmc from it is not supported, it is better to convert the
+				// stats.json emitted to openmetrics in the test itself and upload it to the cluster
+				if t.ExportOpenmetrics() {
+					// Creating a prefix
+					statsFilePrefix := fmt.Sprintf("warehouses=%d/", warehouses)
+
+					// Create buffer for performance metrics
+					perfBuf := bytes.NewBuffer([]byte{})
+					exporter := roachtestutil.CreateWorkloadHistogramExporterWithLabels(t, c, map[string]string{"warehouses": fmt.Sprintf("%d", warehouses)})
+					writer := io.Writer(perfBuf)
+					exporter.Init(&writer)
+					defer func() {
+						if err := exporter.Close(func() error {
+							if _, err = roachtestutil.CreateStatsFileInClusterFromExporterWithPrefix(ctx, t, c, perfBuf, exporter, group.LoadNodes, statsFilePrefix); err != nil {
+								return err
+							}
+							return nil
+						}); err != nil {
+							t.Errorf("failed to close exporter: %v", err)
+						}
+					}()
+
+					if err := getHistogramOpenmetrics(exporter, snapshots, perfBuf); err != nil {
```

Comment:
Rename to `exportOpenMetrics` since the helper isn't just "getting" data? It's essentially performing a conversion and outputting it to `perfBuf`.